.github/workflows/maven-publish.yaml

```diff
@@ -0,0 +1,70 @@
+# This workflow will build a package using Maven and then publish it to GitHub packages when a release is created
+# For more information see: https://github.com/actions/setup-java/blob/main/docs/advanced-usage.md#apache-maven-with-a-settings-path
+
+name: Maven Package
+
+on:
+  pull_request:
+  release:
+    types: [created]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up JDK 17
+        uses: actions/setup-java@v3
+        with:
+          java-version: "17"
+          distribution: "temurin"
+
+      - name: Verify with Maven
+        working-directory: ${{ github.workspace }}/java
+        run: mvn -B verify
+
+  deploy:
+    if: github.ref == 'refs/heads/master'
```

Comment:
this here refers to `'refs/heads/master'` but on line 52, 59 and 65 `'refs/heads/main'` is used; is this an error?

Reply:
```suggestion
    if: github.ref == 'refs/heads/main'
```